exporter/dsn_fix/dsn_fix.go

```diff
@@ -36,7 +39,8 @@ func ClientOptionsForDSN(dsn string) (*options.ClientOptions, error) {
 	username := parsedDsn.User.Username()
 	password, _ := parsedDsn.User.Password()
 	if username != "" || password != "" {
-		clientOptions = clientOptions.SetAuth(options.Credential{Username: username, Password: password})
+		clientOptions.Auth.Username = username
```

Comment:
Just please be sure that setting Username and Password directly is concurrency safe.